java/src/main/java/io/cucumber/cienvironment/CiEnvironmentImpl.java

```diff
@@ -80,28 +78,27 @@ private Git detectGit(Map<String, String> env, Function<Path, Stream<String>> ge
         );
     }
 
-    private String evaluateRevision(Map<String, String> env, Function<Path, Stream<String>> getLines) {
+    private String evaluateRevision(Map<String, String> env) {
         if ("pull_request".equals(env.get("GITHUB_EVENT_NAME"))) {
             if (env.get("GITHUB_EVENT_PATH") == null) {
                 throw new RuntimeException("GITHUB_EVENT_PATH not set");
             }
             Path path = Paths.get(env.get("GITHUB_EVENT_PATH"));
-            return getJsonProperty(path, getLines, "after");
+            try {
+                String json = Files.lines(path).collect(Collectors.joining());
+                Json ob = Json.read(json);
+                try {
+                    return ob.asJsonMap().get("pull_request").asJsonMap().get("head").asJsonMap().get("sha").asString();
+                } catch (RuntimeException e) {
```

Comment:
Is there a dedicated exception to catch?

Reply:
Yeah, but I think I’ll use a shaded minimal-json instead.